main/lsp/requests/code_action.cc

```diff
@@ -260,6 +262,29 @@ unique_ptr<ResponseMessage> CodeActionTask::runRequest(LSPTypecheckerDelegate &t
             action->kind = CodeActionKind::RefactorRewrite;
             action->edit = move(workspaceEdit);
             result.emplace_back(move(action));
+        } else {
+            if (loc.beginPos() != loc.endPos() && config.opts.lspExtractToVariableEnabled) {
```

Comment:
w.r.t. the discussion below of "making a selection in a const prop means that we don't get this refactor anymore"

We could invert this, and say that if there's a selection, the extract to variable refactor takes priority over the move method to new module one.

I'm fine doing that in a follow up PR.